src/core/ioc.py

```diff
@@ -0,0 +1,77 @@
+# structure for the ioc to have just one instance
+
+from src.core.singleton_decorator import singleton
+from src.lib.repositories.impl import *
+from src.api.controllers import *
+
+
+@singleton
+class Ioc:
+
+    _instance_ioc = {}
+
+    def __init__(self):
+
+        self._init_repositories()
+        self._init_controllers()
+
+    def _init_repositories(self):
+        self._instance_ioc[
+            "product_repository_impl"
```

Comment:
It is not a good idea to set `impl` at the end. You should only have 1 source of true for all the registered instances in the pool.

Reply:
copied!